lib/screens/v2/widget_instance/common/base_alert.ex

```diff
@@ -236,23 +200,68 @@ defmodule Screens.V2.WidgetInstance.Common.BaseAlert do
 
   defp get_location_atom(_, _, _), do: :elsewhere
 
+  # Only used to decide "takeover_alert" for bus shelter, bus einks, and prefare screens
   def informs_all_active_routes_at_home_stop?(t) do
-    MapSet.subset?(active_routes_at_stop(t), informed_routes_at_home_stop(t))
+    MapSet.subset?(active_routes_at_stop(t), MapSet.new(informed_routes_at_home_stop(t)))
   end
 
   @spec active_routes_at_stop(t()) :: MapSet.t(route_id())
-  def active_routes_at_stop(t) do
-    t
-    |> SAW.routes_at_stop()
+  def active_routes_at_stop(%{location_context: %{routes: routes}}) do
+    routes
     |> Enum.filter(& &1.active?)
     |> MapSet.new(& &1.route_id)
   end
 
-  @spec informed_routes_at_home_stop(t()) :: MapSet.t(Route.id())
+  def informed_subway_routes(%{screen: %Screen{app_id: app_id}} = t) do
+    t
+    |> informed_entities()
+    |> Enum.map(fn %{route: route} -> route end)
+    # If the alert impacts CR or other lines, weed that out
+    |> Enum.filter(fn e ->
+      Enum.member?(["Red", "Orange", "Green", "Blue"] ++ @green_line_branches, e)
+    end)
+    |> Enum.uniq()
+    |> consolidate_gl(app_id)
+  end
+
+  # GL E-ink consolidates the GL branches to Green Line if there are > 2 branches
+  # We wdant to list all affected branches for the alert and not just the one serving the home stop
```

Comment:
```suggestion
  # We want to list all affected branches for the alert and not just the one serving the home stop
```

Is this comment also true for the pre-fare clause? Seems like it is.

Reply:
Thanks! Shouldn't be

Reply:
I'm not following this comment's meaning then—is it talking about what the `affected_routes` list should contain?

As in, in the places where this function gets called for GL e-ink, we should make sure that `affected_routes` includes all branches affected by the alert, instead of just the affected branches that also serve the home stop?

Reply:
OH! You are totally right. That comment was actually a relic from when I thought the consolidation only happened for GL e-ink, and then later made the clause for Pre-Fare, and didn't integrate the two clauses better. 

This made me realize that I could organize this a little better. Like, the GL clause would work for Pre-Fare if the conditional was `if length(green_routes) > 3`, so I'm going to consolidate a bit